domain/robstoll-lib/atrium-domain-robstoll-lib-jvm/src/main/kotlin/ch/tutteli/atrium/domain/robstoll/lib/creating/pathAssertions.kt

```diff
@@ -1,18 +1,296 @@
 package ch.tutteli.atrium.domain.robstoll.lib.creating
 
 import ch.tutteli.atrium.assertions.Assertion
+import ch.tutteli.atrium.assertions.AssertionGroup
+import ch.tutteli.atrium.assertions.builders.withExplanatoryAssertion
+import ch.tutteli.atrium.assertions.builders.withFailureHintBasedOnDefinedSubject
 import ch.tutteli.atrium.creating.Expect
 import ch.tutteli.atrium.domain.builders.ExpectImpl
+import ch.tutteli.atrium.domain.robstoll.lib.creating.filesystem.Failure
+import ch.tutteli.atrium.domain.robstoll.lib.creating.filesystem.Success
+import ch.tutteli.atrium.domain.robstoll.lib.creating.filesystem.explainForResolvedLink
+import ch.tutteli.atrium.domain.robstoll.lib.creating.filesystem.runCatchingIo
 import ch.tutteli.atrium.reporting.RawString
-import ch.tutteli.atrium.translations.DescriptionBasic.NOT_TO
-import ch.tutteli.atrium.translations.DescriptionBasic.TO
-import ch.tutteli.atrium.translations.DescriptionPathAssertion.EXIST
-import ch.tutteli.niok.exists
-import ch.tutteli.niok.notExists
+import ch.tutteli.atrium.reporting.translating.Translatable
+import ch.tutteli.atrium.translations.DescriptionBasic.*
+import ch.tutteli.atrium.translations.DescriptionPathAssertion.*
+import ch.tutteli.niok.isReadable
+import ch.tutteli.niok.isWritable
+import ch.tutteli.niok.posixFilePersmissions
+import ch.tutteli.niok.readAttributes
+import java.io.IOException
+import java.nio.file.AccessDeniedException
+import java.nio.file.NoSuchFileException
 import java.nio.file.Path
+import java.nio.file.attribute.BasicFileAttributes
+import java.nio.file.attribute.PosixFilePermission
+import java.nio.file.attribute.PosixFilePermission.*
 
 fun <T : Path> _exists(assertionContainer: Expect<T>): Assertion =
-    ExpectImpl.builder.createDescriptive(assertionContainer, TO, RawString.create(EXIST)) { it.exists }
+    ExpectImpl.changeSubject.unreported(assertionContainer) {
```

Comment:
you could move 
```
    ExpectImpl.changeSubject.unreported(assertionContainer) {
        it.runCatchingIo { readAttributes<BasicFileAttributes>() }
    }.let { fileAttributesAssertionContainer ->
```
to an own function an reuse it here and in _existsNot

Reply:
Done.